array_api_tests/pytest_helpers.py

```diff
@@ -483,6 +487,48 @@ def assert_fill(
         assert xp.all(xp.equal(out, xp.asarray(fill_value, dtype=dtype))), msg
 
 
+def scalar_eq(s1: Scalar, s2: Scalar) -> bool:
+    if cmath.isnan(s1):
+        return cmath.isnan(s2)
+    else:
+        return s1 == s2
+
+
+def assert_kw_copy(func_name, x, out, data, copy):
+    """
+    Assert copy=True/False functionality is respected
+
+    TODO: we're not able to check scalars with this approach
```

Comment:
This doesn't look right.
- `astype(copy=True)` should behave like `asarray(copy=True)`.
- `astype(copy=False)` should behave like `asarray(copy=None)`.
- `asarray(copy=False)` has no matching behaviour in `astype`.